pkg/agent/types/knapsack.go

```diff
@@ -0,0 +1,30 @@
+package types
+
+import "go.etcd.io/bbolt"
+
+// Knapsack is an inventory of data and useful services which are used throughout
+// launcher code and are typically valid for the lifetime of the launcher application instance.
+type Knapsack struct {
+	Storage Storage
```

Comment:
`Storage` feels wrong. Launcher has a bunch of different `storage` things under the hood, and I think they should each be a member here.

Reply:
Yeah originally I was thinking the `Storage` abstraction would be helpful when `Knapsack` isn't present, but it's probably unnecessary since we will inject `Knapsack` into plenty of places anyways.

I've moved all the functionality previously in `Storage` up to this top layer now, the storage objects get injected into `Knapsack`, and are immediately accessible from there. Feels better!